gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/files/ProjectFile.java

```diff
@@ -35,31 +37,60 @@ default T overwrite(String text) {
         return (T) this;
     }
 
+    @FormatMethod
+    default T overwrite(@FormatString String text, Object... args) {
+        return overwrite(args.length > 0 ? text.formatted(args) : text);
+    }
+
     default T append(String text) {
```

Comment:
Ideally we'd also make an errorprone to check that people don't do:

```java
file.append("foo %s".formatted(3))
file.append(String.format("foo %s", 3))
```

...and force them to do:

```java
file.append("foo %s", 3)
```

...instead to ensure people get the syntax highlighting. But I need to find the time to get #221 in first, so later PR.